### PR TITLE
fix(suiteheader): icon vertical alignment

### LINKED ISSUE #3694: [SuiteHeader] icons not aligned vertically and 1 px horizontal color change
### What package is this for?

- [x] React
- [ ] Angular

### Describe the bug

UI Shell header - icons not aligned vertically and 1 px horizontal color change

### To Reproduce

1. Go to https://next.carbon-addons-iot-react.com/?path=/story/1-watson-iot-ui-shell-suiteheader-multi-workspace--header-with-custom-action-items
2. Look at the UI Shell Header
  A. Right-most icons are not aligned properly (should be vertically centered)  
  B. 1px grey line across bottom of 48 px header is black under some of the buttons

### Expected behavior

- All header icons should be aligned properly (help and user icons are 2 px too high)
- Bottom pixel of header should have a consistent color (1px grey)

### Environment/versions:

- OS: MacOS, Windows
- Browser: Chrome
- `carbon-addons-iot-react` version: 2.151.1

### Additional context

Screenshots:
Show icon misalignment in RB and PAL and the bottom border grey overlayed with black for only some buttons. (In the third attachment where the purple dimension lines are shown.)

Watson-iot-ui-shell-suiteheader - with markup

Please provide any additional relevant information below:

- Icon alignment and different bottom border issue also seen in PAL storybook
- bx--header has a unique color for the bottom border:
border-bottom: 1px solid #393939;
  border-bottom-color: rgb(57, 57, 57);
  border-bottom-style: solid;
  border-bottom-width: 1px;
... but some buttons or menus seem to overlay it with the background color

### Specific timeline issues / requests

NA

Issue is originally reported by Michael Zuliani (IBM Engineering Lifecycle Management)

### PR DESCRIPTION
**Pull request by erzhan-temir-mamyrov**
Closes #3694

**Summary**

- Align icon buttons vertically in SuiteHeader
- Remove overlapping background in order to display border-bottom

**Change List (commits, features, bugs, etc)**

- Update styles for SuiteHeader

**Acceptance Test (how to verify the PR)**

- Go to [this story](https://deploy-preview-3708--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-ui-shell-suiteheader-multi-workspace--header-with-custom-action-items)
- Verify that all icons aligned vertically
- Verify that header has persistent border-bottom

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

**Review by davidicus — APPROVED** [review]

File: packages/react/src/components/SuiteHeader/_suite-header.scss [hunk 1]
```diff
@@ -324,6 +324,17 @@
       display: flex;
       align-items: center;
     }
+
+    .#{$prefix}--header-action-btn.#{$prefix}--btn--primary {
+      background-color: unset;
+    }
+
+    & > *:nth-last-child(2),
```
Comment (davidicus) [inline packages/react/src/components/SuiteHeader/_suite-header.scss:332]: @erzhan-temir-mamyrov Is there a way to make this selector more broad?

Comment (erzhan-temir-mamyrov) [reply packages/react/src/components/SuiteHeader/_suite-header.scss:332]: @davidicus Thanks for commenting on this! You mean to make this selector more specific?

Comment (davidicus) [reply packages/react/src/components/SuiteHeader/_suite-header.scss:332]: @erzhan-temir-mamyrov Actually less specific. Just wondering if this is broad enough to catch all instances.

Comment (davidicus) [reply packages/react/src/components/SuiteHeader/_suite-header.scss:332]: If this is pretty locked down then we can approve as is.

Comment (erzhan-temir-mamyrov) [reply packages/react/src/components/SuiteHeader/_suite-header.scss]: @davidicus I added class selectors to make it more global.

**Review by davidicus — APPROVED** [review]

**Pull request merged**